src/pipeline.cc

```diff
@@ -279,23 +279,6 @@ class PipelineWorker : public Nan::AsyncWorker {
         }
         xfactor = static_cast<double>(shrunkOnLoadWidth) / static_cast<double>(targetResizeWidth);
         yfactor = static_cast<double>(shrunkOnLoadHeight) / static_cast<double>(targetResizeHeight);
-        xshrink = std::max(1, static_cast<int>(floor(xfactor)));
-        yshrink = std::max(1, static_cast<int>(floor(yfactor)));
-        xresidual = static_cast<double>(xshrink) / xfactor;
-        yresidual = static_cast<double>(yshrink) / yfactor;
-        if (
```

Comment:
I'm slightly worried that removing the `swap` here will break something but the tests that were added for the original bug #387 continue to pass. There's probably an edge case that, if ever found, we can add the fix and another test for later :)

Reply:
Thats removed because xresidual is not used after that point. Its used only before for the shrink_on_load edge case.